internal/common/labels.go

```diff
@@ -38,15 +39,20 @@ const (
 // Instance wide labels will be taken from the request if available
 func AddAppLabels(requestInstance *ssp.SSP, name string, component AppComponent, obj client.Object) client.Object {
```

Comment:
Can you change the type of the last argument in this function to `metav1.Object` interface? Then you can use it for the deployment templates, and you can remove the new function `AddAppLabelsToMeta()`.

Also the return value of this function is not used anywhere (except tests), so you can remove it if you want.

Reply:
Thanks. I wasn't aware of it.